src/mixins/ui/lasso-tool-ui.js

```diff
@@ -52,6 +138,23 @@ class ShapeHandler {
       typeof this.chart.useLonLat === "function" && this.chart.useLonLat()
   }
 
+  /**
+   * @returns {LassoToolSetTypes}
+   */
+  getLassoToolType() {
+    assert(
+      false,
+      `${ShapeHandler.name}::getLassoToolType() needs to be overridden by derived class`
+    )
+  }
+
+  fireEvent(event_signal, event_obj = {}) {
+    this.drawEngine.fire(event_signal, {
+      ...event_obj,
+      lasso_tool_type: this.getLassoToolType()
```

Comment:
adds the `lasso_tool_type` property to the event object that can be read by callbacks